### PR TITLE
some refinement

### PR DESCRIPTION
**Pull request by wongsyrone**

### CODE REVIEW

File: src/proto/socks5address.cpp [hunk 1]
```diff
@@ -22,33 +22,40 @@
...
             }
             break;
         }
         case DOMAINNAME: {
-            if (data.length() >= (unsigned int)(unsigned char)data[1] + 4) {
-                address = data.substr(2, data[1]);
-                port = (uint8_t(data[data[1] + 2]) << 8) | uint8_t(data[data[1] + 3]);
-                return data[1] + 4;
+            uint8_t domain_len = data[1];
+            if (domain_len < 1 || domain_len > 255) {
```
Comment (GreaterFire) [inline src/proto/socks5address.cpp]: It's impossible for an 8-bit number to exceed 255.
```suggestion
            if (domain_len == 0) {
```

File: src/proto/udppacket.cpp [hunk 2]
```diff
@@ -21,17 +21,22 @@
...
 using namespace boost::asio::ip;
 
-int UDPPacket::parse(const string &data) {
-    int address_len = address.parse(data);
-    if (address_len == -1 || data.length() < (unsigned int)address_len + 2) {
-        return -1;
+bool UDPPacket::parse(const string &data, size_t &udp_packet_len) {
+    if (data.length() <= 0) {
+        return false;
+    }
```
Comment (GreaterFire) [inline src/proto/udppacket.cpp:27]: I don't think this is necessary, since it'll fail the `address.parse` call.

**Review by GreaterFire — APPROVED** [review]
LGTM except for commented code. Is 9fb84c001161f581d52e1229a9f9ebf5304b8f3f solely for code readability?

File: src/proto/udppacket.cpp (hunk 2 above)
Comment (wongsyrone) [reply src/proto/udppacket.cpp:27]: Just to make it clear.

Comment (GreaterFire) [reply src/proto/udppacket.cpp:27]: Sure.

**Comment by GreaterFire** [comment]
LGTM. Will merge after CI.

**Comment by wongsyrone** [comment]
> Is [9fb84c0](https://github.com/trojan-gfw/trojan/commit/9fb84c001161f581d52e1229a9f9ebf5304b8f3f) solely for code readability?

Yes, it doesn't really fix something, just to make the logic obvious to everyone.

**Pull request merged**